TGM/src/main/java/com/minehut/tgm/modules/infection/InfectionModule.java

```diff
@@ -0,0 +1,253 @@
+                playerContext.getPlayer().setGameMode(GameMode.ADVENTURE);
+            });
+        }
+
+        new BukkitRunnable() {
+            @Override
+            public void run() {
+                if (match.getMatchStatus().equals(MatchStatus.MID) && teamManager.getTeamById("humans").getMembers().size() != 0) {
+                    TGM.get().getMatchManager().endMatch(teamManager.getTeamById("humans"));
+                }
+            }
+        }.runTaskLater(TGM.get(), length * 60 * 20);
+
+
+    }
+
+//    @EventHandler
+//    public void onPlayerDeath(PlayerDeathEvent event) {
+//        // If the player isn't on the spectator team, they must be either human or infected.
+//        if (!teamManager.getTeam(event.getPlayer()).getId().equals("spectators")) {
+//
+//            // "infected" = zombie team; "humans" = human team;
+//            if (teamManager.getTeam(event.getPlayer()).getId().equals("humans")) {
+//                // Now to determine if the user got infected by a player, or died because of anything else (still infected).
+//                if (event instanceof PlayerDeathByPlayerEvent) {
+//                    broadcastMessage(String.format("%s%s &7has been infected by %s%s",
+//                            teamManager.getTeam(event.getPlayer()).getColor(),
+//                            event.getPlayer().getName(),
+//                            teamManager.getTeam(((PlayerDeathByPlayerEvent) event).getCause()).getColor(),
+//                            ((PlayerDeathByPlayerEvent) event).getCause().getName()));
+//
+//                } else {
+//                    broadcastMessage(String.format("%s%s &7has been taken by the environment",
+//                            teamManager.getTeam(event.getPlayer()).getColor(),
+//                            event.getPlayer().getName()));
+//                }
+//
+//                infect(event.getPlayer());
+//            } else if (teamManager.getTeam(event.getPlayer()).getId().equalsIgnoreCase("infected")) {
+//                if (event instanceof PlayerDeathByPlayerEvent) {
+//                    if (teamManager.getTeam(((PlayerDeathByPlayerEvent) event).getCause()).getId().equalsIgnoreCase("infected")) {
+//                        return;
+//                    }
+//                    broadcastMessage(String.format("%s%s &7has been slain by %s%s",
+//                            teamManager.getTeam(event.getPlayer()).getColor(),
+//                            event.getPlayer().getName(),
+//                            teamManager.getTeam(((PlayerDeathByPlayerEvent) event).getCause()).getColor(),
+//                            ((PlayerDeathByPlayerEvent) event).getCause().getName()
+//                            ));
+//                }
+//            }
+//
+//        }
+//    }
+
+    @EventHandler
+    public void onDamage(PlayerDamageEvent event) {
+        if (event.getEntity().getHealth() - event.getDamage() < 1) {
```

Comment:
Cancelling the death event will stop other modules from being able to track kills/deaths. An example side effect of this is kill stat tracking not working for Infection.